### PR TITLE
Gateway can process list backup request

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

On receiving list backup request, gateway send the request to all partitions and aggregate the responses. Each partition returns a list of backups. An aggregated status of each backup must be calculated from the statuses returned from each partitions. This is handled in `BackupRequestHandler` and it returns a list of backups.

## Related issues

related #10849 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
Thanks, LGTM :+1:

File: gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupListRequest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Override
+  protected BackupListResponse toResponseDto(final DirectBuffer buffer) {
+    return response;
+  }
+
+  @Override
+  public String getType() {
+    return "backup#list";
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupListRequest.java:86]: :+1: I will change it.

File: gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupRequestHandler.java [hunk 2]
```diff
@@ -69,6 +73,81 @@ public CompletionStage<BackupStatus> getStatus(final long backupId) {
             });
   }
 
+  @Override
+  public CompletionStage<List<BackupStatus>> listBackups() {
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply gateway/src/main/java/io/camunda/zeebe/gateway/admin/backup/BackupRequestHandler.java:77]: :+1: This looks better. I will apply this change.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**